libs/common/GarnetException.cs

```diff
@@ -33,5 +34,13 @@ public GarnetException(string message) : base(message)
         public GarnetException(string message, Exception innerException) : base(message, innerException)
         {
         }
+
+        /// <summary>
+        /// Throw helper that throws a GarnetException.
+        /// </summary>
+        /// <param name="message">Exception message.</param>
+        [DoesNotReturn]
+        public static void Throw(string message) =>
```

Comment:
nit; there is a warning in the inherited method from RespParsingException. 
https://github.com/microsoft/garnet/blob/951cf82c120d4069e940e832db03bfa018c688ea/libs/common/Parsing/RespParsingException.cs#L82

Can you explicitly override the inherited method as below
```csharp 
        [DoesNotReturn]
        public static new void Throw(string message) =>
            throw new RespParsingException(message);
```